nav2_bt_navigator/include/nav2_bt_navigator/bt_navigator.hpp

```diff
@@ -84,11 +84,23 @@ class BtNavigator : public nav2_util::LifecycleNode
    */
   nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
 
+  /**
+   * @brief Load Naigator plugins
+   * @param plugin_names bt plugin libaray names
+   * @return SUCCESS or FAILURE 
+   */
+  bool loadNavigatorPlugins(std::vector<std::string> plugin_names);
+
   // To handle all the BT related execution
-  std::unique_ptr<nav2_bt_navigator::Navigator<nav2_msgs::action::NavigateToPose>> pose_navigator_;
-  std::unique_ptr<nav2_bt_navigator::Navigator<nav2_msgs::action::NavigateThroughPoses>>
-  poses_navigator_;
-  nav2_bt_navigator::NavigatorMuxer plugin_muxer_;
+  // Navigator plugin
+  pluginlib::ClassLoader<nav2_core::NavigatorBase> navigator_class_loader_;
+  std::vector<pluginlib::UniquePtr<nav2_core::NavigatorBase>> navigators_;
+  std::vector<std::string> default_navigator_ids_;
+  std::vector<std::string> default_navigator_types_;
+  std::vector<std::string> navigator_ids_;
+  std::vector<std::string> navigator_types_;
```

Comment:
I don't think `navigator_types_` needs to be stored. You only use it for logging, you can just create a `std::string` in that function to hold the current one

Reply:
I agree with you. so I removed at this commit.
https://github.com/ladianchad/navigation2/commit/59ced2b08eb0ce4a26f7c7afd7994c16e6c0dec8